config/pmd.xml

```diff
@@ -372,7 +372,8 @@
       | //ClassDeclaration[@SimpleName='SarifLogger']//MethodDeclaration[@Name='escape']
       | //ClassDeclaration[@SimpleName='LeftCurlyCheck'
         or @SimpleName='FinalLocalVariableCheck'
-        or @SimpleName='NPathComplexityCheck']
+        or @SimpleName='NPathComplexityCheck'
+        or @SimpleName='RedundantModifierCheck']
```

Comment:
Suppression was added, but no reason why.

Reply:
It goes under the already existing reason, comment above the suppressions.

> The methods visitToken/leaveToken are a big SWITCH block with a number of IF blocks.
             If we split the block to several methods it will damage the readability.

Reply:
The one which lists
> Main, PackageObjectFactory, RequireThisCheck, HandlerFactory,
VariableDeclarationUsageDistanceCheck,

?

Reply:
comment 
```
      <!-- Main, PackageObjectFactory, RequireThisCheck, HandlerFactory,
             VariableDeclarationUsageDistanceCheck,
             are big and should not be split without a good reason.  (This is for classReportLevel)
             
             The methods visitToken/leaveToken are a big SWITCH block with a number of IF blocks.
             If we split the block to several methods it will damage the readability. (This is for methodReportLevel, it doesn't list specific classes, it just says a method name which is visit or leave token)
```
supression
```
//ClassDeclaration[@SimpleName='Main'
        or @SimpleName='PackageObjectFactory' or @SimpleName='RequireThisCheck'
        or @SimpleName='VariableDeclarationUsageDistanceCheck'
        or @SimpleName='HandlerFactory']    ( First comment)
        ....
        
              | //ClassDeclaration[@SimpleName='LeftCurlyCheck'
        or @SimpleName='FinalLocalVariableCheck'
        or @SimpleName='NPathComplexityCheck'
        or @SimpleName='RedundantModifierCheck']
             //MethodDeclaration[@Name='visitToken']  (Second comment)
```

Reply:
I am ok with suppression

Reply:
Suppression is not the issue. We declare with the suppression why we are suppressing it.

We add in the name of the class and the reason. Since the class name was never added, I can't tell why we are suppressing it, if it is agreeable, and what we would need to do in the future if we ever wanted to resolve it. The comments before it all have class names listed as we pointed to.

Reply:
From what I see in the comment above suppression I think it was agreed on before that we can suppress violations in visit/leave Token if it has a big switch block regardless of the name of the class.

I can add the name of this check if you want but we didn't do this before with this specific case (visitToken method), if the suppressed method was anything other than this I would feel different.

Reply:
Do you want comment like this?

```

             The methods visitToken/leaveToken "in LeftCurlyCheck,  FinalLocalVariableCheck, NPathComplexityCheck and RedundantModifierCheck are a big SWITCH block with a number of IF blocks.
             If we split the block to several methods it will damage the readability.
```

Reply:
Ok, I changed my mind and am going to drop this.

> visit/leave Token if it has a big switch block regardless of the name of the class.

I know see what you mean by this. I don't think the way we organized these reasons here is very straightforward, but you are correct that we give a blanket statement based on method name.